awx/ui_next/src/screens/Inventory/InventorySources/InventorySourceList.jsx

```diff
@@ -0,0 +1,155 @@
+  const { inventoryType, id } = useParams();
+  const { search } = useLocation();
+
+  const {
+    isLoading,
+    error,
+    result: { sources, sourceCount, sourceChoices, sourceChoicesOptions },
+    request: fetchSources,
+  } = useRequest(
+    useCallback(async () => {
+      const params = parseQueryString(QS_CONFIG, search);
+      const results = await Promise.all([
+        InventoriesAPI.readSources(id, params),
+        InventorySourcesAPI.readOptions(),
+      ]);
+
+      return {
+        sources: results[0].data.results,
+        sourceCount: results[0].data.count,
+        sourceChoices: results[1].data.actions.GET.source.choices,
+        sourceChoicesOptions: results[1].data.actions,
+      };
+    }, [id, search]),
+    {
+      sources: [],
+      sourceCount: 0,
+    }
+  );
+
+  useEffect(() => {
+    fetchSources();
+  }, [fetchSources]);
+
+  const { selected, isAllSelected, handleSelect, setSelected } = useSelected(
+    sources
+  );
+
+  const {
+    isLoading: isDeleteLoading,
+    deleteItems: handleDeleteSources,
+    deletionError,
+    clearDeletionError,
+  } = useDeleteItems(
+    useCallback(async () => {
+      return Promise.all(
+        selected.map(({ id: sourceId }) =>
+          InventorySourcesAPI.destroy(sourceId)
+        ),
+        []
+      );
+    }, [selected]),
+    {
+      fetchItems: fetchSources,
+      allItemsSelected: isAllSelected,
+      qsConfig: QS_CONFIG,
+    }
+  );
+
+  const handleDelete = async () => {
+    await handleDeleteSources();
+    setSelected([]);
+  };
+
+  const canAdd =
+    sourceChoicesOptions &&
+    Object.prototype.hasOwnProperty.call(sourceChoicesOptions, 'POST');
+  const detailUrl = `/inventories/${inventoryType}/${id}/sources/`;
+  return (
+    <>
+      <PaginatedDataList
+        contentError={error || deletionError}
+        hasContentLoading={isLoading || isDeleteLoading}
+        items={sources}
+        itemCount={sourceCount}
+        pluralizedItemName={i18n._(t`Sources`)}
+        qsConfig={QS_CONFIG}
+        renderToolbar={props => (
+          <DatalistToolbar
+            {...props}
+            showSelectAll
+            isAllSelected={isAllSelected}
+            onSelectAll={isSelected =>
+              setSelected(isSelected ? [...sources] : [])
+            }
+            qsConfig={QS_CONFIG}
+            additionalControls={[
+              ...(canAdd
+                ? [<ToolbarAddButton key="add" linkTo={`${detailUrl}add`} />]
+                : []),
+              <ToolbarDeleteButton
+                key="delete"
+                onDelete={handleDelete}
+                itemsToDelete={selected}
+                pluralizedItemName={i18n._(t`Inventory Source`)}
+              />,
+            ]}
+          />
+        )}
+        renderItem={inventorySource => {
+          sourceChoices.forEach(([scMatch, scLabel]) => {
+            if (inventorySource.source === scMatch) {
+              inventorySource.source = scLabel;
```

Comment:
We should avoid mutating this data object if possible. Instead we could create a new "label" prop and update the source column via that prop.